wit/crud.wit

```diff
@@ -2,26 +2,27 @@
 /// 
 /// A CRUD operation is an operation that acts on a single key-value pair.
 /// 
-/// The value in the key-value pair is defined as a `u8` byte array and the intention
-/// is that it is the common denominator for all data types defined by different
-/// key-value stores to handle data, ensuring compatibility between different
-/// key-value stores. Note: the clients will be expecting serialization/deserialization overhead
-/// to be handled by the key-value store. The value could be a serialized object from
-/// JSON, HTML or vendor-specific data types like AWS S3 objects.
+/// The value in the key-value pair is defined as a `u8` byte array and the intention is that it is
+/// the common denominator for all data types defined by different key-value stores to handle data,
+/// ensuring compatibility between different key-value stores. Note: the clients will be expecting
+/// serialization/deserialization overhead to be handled by the key-value store. The value could be
+/// a serialized object from JSON, HTML or vendor-specific data types like AWS S3 objects.
 /// 
-/// Data consistency in a key value store refers to the guarentee that once a 
-/// write operation completes, all subsequent read operations will return the
-/// value that was written.
+/// Data consistency in a key value store refers to the guarantee that once a write operation
+/// completes, all subsequent read operations will return the value that was written.
 /// 
-/// The level of consistency in readwrite interfaces is **eventual consistency**,
-/// which means that if a write operation completes successfully, all subsequent
-/// read operations will eventually return the value that was written. In other words,
-/// if we pause the updates to the system, the system eventually will return 
-/// the last updated value for read.
+/// The level of consistency in this interface is **eventual consistency**, which means that if a
+/// write operation completes successfully, all subsequent read operations will eventually return
+/// the value that was written. In other words, if we pause the updates to the system, the system
+/// eventually will return the last updated value for read.
 interface key-value {
   use types.{store, error};
+  
+  /// The crud resource contains the common create, read, update, and delete operations for a
+  /// key-value store
   resource crud {
-	open: static func(store: borrow<store>) -> crud;
+    /// Create a new crud resource, using the given store to access the data
+	  create: static func(store: borrow<store>) -> crud;
```

Comment:
I was talking to Joel about these constructors and he mentioned 2 things:
1. If we are not returning errors, we should use a WIT constructor
2. He pointed out the following: what if your KV store implements crud and batch but not atomic? We do need to return an error here for that case

This may be a discussion to have outside of this cleanup PR though

Reply:
If a KV store doesn't implement one of the interfaces, wouldn't the component fail to run because of an unsatisfied import?

Reply:
I was imagining a world here a host component would "implement" all the interfaces so as to be compatible, but the implementation for some may just be "unimplemented/unsupported". Requiring all KV stores to support all the interfaces seems like somewhat of a lofty goal

Reply:
Hmmm, this is definitely an interesting discussion point. I think my opinion right now is "if you don't support that interface, don't export it"

Reply:
But also, to provide a counter argument to myself, some people might want to implement 90% of wasi-cloud-core and be compliant. Not sure if that should be a supported thing or not